src/python/CMakeLists.txt

```diff
@@ -78,6 +85,67 @@ install(FILES "${OpenVINOGenAI_SOURCE_DIR}/LICENSE"
 
 # wheel_genai component is used for wheel generation in pyproject.toml.
 # Exclude wheel_genai from normal packaging because there's pygenai_X_Y component for that.
-install(TARGETS openvino_genai py_openvino_genai
+install(TARGETS openvino_genai ${TARGET_NAME}
         LIBRARY DESTINATION openvino_genai COMPONENT wheel_genai EXCLUDE_FROM_ALL
         RUNTIME DESTINATION openvino_genai COMPONENT wheel_genai EXCLUDE_FROM_ALL)
+
+# Generate or check .pyi stub files generated by pybind11-stub
+
+set(pyproject_toml "${OpenVINOGenAI_SOURCE_DIR}/pyproject.toml")
+file(STRINGS ${pyproject_toml} pybind11_stubgen_dep REGEX "pybind11-stubgen")
+
+if(pybind11_stubgen_dep MATCHES "pybind11-stubgen==[0-9\.]+")
+    set(pybind11_stubgen_dep "${CMAKE_MATCH_0}")
+else()
+    message(FATAL_ERROR "Internal error: failed to parse pybind11-stubgen version from from '${pyproject_toml}'")
+endif()
+
+if(OpenVINODeveloperPackage_FOUND)
+    set(requirements_file "${OpenVINOGenAI_SOURCE_DIR}/requirements-build.txt")
+
+    ov_check_pip_package(REQUIREMENT ${pybind11_stubgen_dep}
+                         RESULT_VAR pybind11_stubgen_AVAILABLE
+                         WARNING_MESSAGE "Please, install ${pybind11_stubgen_dep} if you plan to develop Python OpenVINO GenAI API"
+                         MESSAGE_MODE WARNING)
+elseif(DEFINED PY_BUILD_CMAKE_PACKAGE_NAME)
+    # in case of wheel build, pybind11-stubgen is always available via pyproject.toml's build-system
+    set(pybind11_stubgen_AVAILABLE ON)
+endif()
+
+if(pybind11_stubgen_AVAILABLE)
+    if(DEFINED ENV{CI} OR DEFINED ENV{TF_BUILD} OR DEFINED ENV{JENKINS_URL})
```

Comment:
You need to set one of these in precommit build

Reply:
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables